waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -89,11 +90,20 @@ func (wakuM *WakuMetadata) getClusterAndShards() (*uint32, []uint32, error) {
 	return &u32ClusterID, shards, nil
 }
 
+func isProtocolNotSupported(err error) bool {
+	notSupportedErr := multistream.ErrNotSupported[libp2pProtocol.ID]{}
+	return errors.Is(err, notSupportedErr)
+}
+
 func (wakuM *WakuMetadata) Request(ctx context.Context, peerID peer.ID) (*protocol.RelayShards, error) {
 	logger := wakuM.log.With(logging.HostID("peer", peerID))
 
 	stream, err := wakuM.h.NewStream(ctx, peerID, MetadataID_v1)
 	if err != nil {
+		if isProtocolNotSupported(err) {
+			return nil, nil
```

Comment:
This return type is misleading for this function as the stream opening failed and we are still returning err non nil.
Better to add such behvaiour in comments so that caller is aware.

Reply:
With last commit, this observation no longer applies